cmd/neo4j-mcp/main.go

```diff
@@ -25,14 +27,35 @@ func main() {
 		log.Fatalf("Failed to load configuration: %v", err)
 	}
 
-	// Create and configure the MCP server
-	mcpServer, err := server.NewNeo4jMCPServer(Version, cfg)
+	// Initialize Neo4j driver
+	driver, err := neo4j.NewDriverWithContext(cfg.URI, neo4j.BasicAuth(cfg.Username, cfg.Password, ""))
 	if err != nil {
-		log.Fatalf("Failed to create MCP server: %v", err)
+		log.Fatalf("Failed to create Neo4j driver: %v", err)
 	}
 
 	// Gracefully handle shutdown
 	ctx := context.Background()
+	defer func() {
+		if err := driver.Close(ctx); err != nil {
+			log.Fatalf("Error closing driver: %v", err)
+		}
+	}()
+
+	// Verify database connectivity
+	if err := driver.VerifyConnectivity(ctx); err != nil {
```

Comment:
We move the driver to the main, which helps create a cleaner interface on the internals, and we can still drop in a mock driver for unit tests.